### PR TITLE
Updated sdk development guide

### LINKED ISSUE #351: Update SDK development guide
The guide https://nemtech.github.io/guidelines/sdk-development.html should be reviewed and updated according the last changes in how clients and catbuffer libraries are generated.

**Sections to review:**

- [ ] API Wrapper
- [ ] Models
- [ ] Transaction Serialization

### PR DESCRIPTION
**Pull request by dgarcia360**
Closes #351

### CODE REVIEW

File: source/guidelines/sdk-development.rst [hunk 1]
```diff
@@ -64,215 +68,142 @@ Reactive
...
+
+   git clone https://github.com/nemtech/catbuffer-generators
+
+2. Install the package requirements.
+
+.. code-block:: bash
+
+   pip install -r requirements.txt
+
+3. Clone the catbuffer repository inside the ``catbuffer-generators`` folder.
```
Comment (fboucquez) [inline source/guidelines/sdk-development.rst]: You can avoid this step by cloning catbuffer-generators with --recursive

`git clone --recursive git@github.com:nemtech/catbuffer-generators.git`

File: source/guidelines/sdk-development.rst [hunk 2]
```diff
@@ -64,215 +68,142 @@ Reactive
...
+
+.. code-block:: bash
+
+   git clone https://github.com/nemtech/catbuffer
+
+4. Generate code for all the schemas running the following command under the ``catbuffer-generators`` directory, replacing ``cpp_builder`` for the targeted programming language.
+
+.. code-block:: bash
+
+   python ../scripts/generate_all.sh cpp_builder
```
Comment (fboucquez) [inline source/guidelines/sdk-development.rst]: Current generate_all.sh doesn't generate all the transactions at once creating incomplete files like `EntityTypeDto`.

I would recommend following the specific language generate script like:

https://github.com/nemtech/catbuffer-generators/blob/master/scripts/generate_java.sh
https://github.com/nemtech/catbuffer-generators/blob/master/scripts/generate_typescript.sh

Where the script also builds, tests and deploys the artifact for the given language.

The script should be added to [travis](https://github.com/nemtech/catbuffer-generators/blob/master/.travis.yml)

File: source/guidelines/sdk-development.rst [hunk 3]
```diff
@@ -281,14 +212,30 @@ KeyPair and Cryptographic functions
 
 Implementing cryptographic functions is required to sign transactions.
 
-Example: `core/crypto <https://github.com/nemtech/nem2-sdk-java/tree/master/sdk-core/src/main/java/io/nem/core/crypto>`_
+Example: `core/crypto <https://github.com/nemtech/symbol-sdk-typescript-javascript/tree/master/src/core/crypto>`_
```
Comment (fboucquez) [inline source/guidelines/sdk-development.rst]: Creators of SDKs should pay special attention to the [vector](https://github.com/nemtech/test-vectors) test. They should create unit tests using the files as input and expected outputs.

Examples of vector:

- https://github.com/nemtech/symbol-sdk-java/blob/master/sdk-core/src/test/java/io/nem/symbol/core/crypto/KeyPairVectorTester.java
- https://github.com/nemtech/symbol-sdk-java/blob/master/sdk-core/src/test/java/io/nem/symbol/core/crypto/DsaSignerVectorTester.java
- https://github.com/nemtech/symbol-sdk-typescript-javascript/blob/master/test/core/crypto/keyPair.spec.ts#L88

Can we extend this section with the vector tests?

File: source/guidelines/sdk-development.rst [hunk 4]
```diff
@@ -64,215 +68,142 @@ Reactive
...
-    }
-
-5. Copy the generated files into the `nem2-sdk infrastructure folder <https://github.com/nemtech/nem2-sdk-typescript-javascript/tree/master/src/infrastructure>`_.
-
-6. Drop the generated client classes and implement them using the `Repository pattern <https://martinfowler.com/eaaCatalog/repository.html>`_ returning `Observables <https://en.wikipedia.org/wiki/Observer_pattern>`_ of `ReactiveX <http://reactivex.io/>`_.
-
-Example of a Repository and HTTP implementation:
+    openapi-generator generate -i ./openapi3.yml -g typescript-node -o ./symbol-ts-sdk/ && rm -R symbol-ts-sdk/test
+
+4. The generated lib are normally published into a central repository (e.g. maven, npm). The SDKs depend on those libraries like any other third party dependency. To automate the deployment of the packages, including the generator for the selected programming language in the `symbol-openapi-generator <https://github.com/nemtech/symbol-openapi-generator>`_ project.
```
Comment (fboucquez) [inline source/guidelines/sdk-development.rst]: Should we mention that Travis is automatically upgrading/releasing new versions of the generated clients when the open api spec is updated? Currently, one of us updates the open api, everybody gets the new clients deployed into the repos. The creator of the new SDK should follow the same pattern and get the same benefits. We can then send a Slack ping saying that there is a new version of the open api clients, dev should update their sdks

File: source/guidelines/sdk-development.rst (hunk 3 above)
Comment (rg911) [reply source/guidelines/sdk-development.rst]: Regarding the recent server updates:
- SDKs are now using standard `tweetnacl` (ed2559) for keyPair generation, address derivation (from public key) and signings. Keypairs are based on `tweetnacl` 64 bytes secretKey (public + private) using `sha512`. 
- Signatures also get generated using sha512. In the latest SDK updates we use `tweetnacl` detached mode for signing. 
- Please be noded that we are still using a custom script for `SharedKey` derivation in the SDK. This is due to the ED2559 Mult function (Edward Curve) is used as an internal function which is not exposed in `tweetnacl`.  There's a discussion around this topic whether to change on the server to adapt the exposed mult function. So, keep eyes on future announcement.

File: source/guidelines/sdk-development.rst (hunk 2 above)
Comment (dgarcia360) [reply source/guidelines/sdk-development.rst]: Ideally, ``generate_all.sh`` should generate all the tranasctions. This note will really helpful for the ``catbuffer-generator`` README.

**Pull request merged**